### PR TITLE
Fix route conflicts for the auto-save and revisions endpoints of the wp_template and wp_template_part post types

### PR DESCRIPTION
**Pull request by anton-vlasenko**
### What and why?

Fixes https://core.trac.wordpress.org/ticket/56922.

This PR addresses route conflicts between `WP_REST_Templates_Controller` and both `WP_REST_Revisions_Controller` and `WP_REST_Autosaves_Controller`.

When making REST requests to the following endpoints:
- `wp/v2/templates/<parent_post_id>/revisions`
- `wp/v2/templates/<parent_post_id>/autosaves`
- `wp/v2/template-parts/<parent_post_id>/revisions`
- `wp/v2/template-parts/<parent_post_id>/autosaves`

the `WP_REST_Templates_Controller::get_item()` method  is incorrectly matched as the callback method to process these requests. This misidentification arises from the regular expression that defines the problematic route in the `WP_REST_Templates_Controller` class, specifically:
`/(?P<id>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)`.

In actuality, the aforementioned requests should be handled by the `WP_REST_Revisions_Controller::get_items()` and `WP_REST_Autosaves_Controller::get_items()` callbacks, depending on whether it's a `revisions` or `autosaves` request.

### How

This PR modifies the order in which the routes for endpoints in `WP_REST_Templates_Controller`, `WP_REST_Revisions_Controller`, and `WP_REST_Autosaves_Controller` are registered.

Routes in `WP_REST_Revisions_Controller` and `WP_REST_Autosaves_Controller` are now registered prior to those in `WP_REST_Templates_Controller`.

This ensures the correct callback is matched, preventing `WP_REST_Templates_Controller::get_item()` from being mistakenly matched for the endpoints listed above.

However, this change is limited to the `wp_template` and `wp_template_parts` post types to mitigate the risk of introducing new bugs.
Snímek obrazovky 2023-09-13 v 1 27 44

Snímek obrazovky 2023-09-13 v 1 28 13

### CODE REVIEW

**Comment by spacedmonkey** [comment]
Why register late? Why not register all post type controller like this?

**Comment by anton-vlasenko** [comment]
> Why register late? Why not register all post type controller like this?

I've replied in the ticket: https://core.trac.wordpress.org/ticket/56922#comment:13
Thank you.

File: tests/qunit/fixtures/wp-api-generated.js [hunk 1]
```diff
@@ -5074,6 +5074,338 @@ mockedApiResponse.Schema = {
                 }
             ]
         },
+        "/wp/v2/templates/(?P<parent>[\\d]+)/revisions": {
```
Comment (anton-vlasenko) [inline tests/qunit/fixtures/wp-api-generated.js]: The changes in the `wp-api-generated.js` file are auto-generated, so probably it makes no sense to review them.

**Comment by anton-vlasenko** [comment]
Closed in favor of https://github.com/WordPress/wordpress-develop/pull/3533.

**Pull request closed without merging**